omniledger/service/service_test.go

```diff
@@ -372,6 +372,75 @@ func TestService_StateChange(t *testing.T) {
 	require.Equal(t, latest, int64(n-1))
 }
 
+func TestService_DarcEvolutionFail(t *testing.T) {
+	s := newSer(t, 1, testInterval)
+	defer s.local.CloseAll()
+	defer closeQueues(s.local)
+
+	d2 := s.darc.Copy()
+	require.Nil(t, d2.EvolveFrom(s.darc))
+
+	// first we create a bad request, i.e., with an invalid version number
+	d2.Version = 11
+	resp := s.testDarcEvolution(t, *d2)
+
+	// parse the darc
+	require.True(t, resp.Proof.InclusionProof.Match())
+	_, vs, err := resp.Proof.KeyValue()
+	require.Nil(t, err)
+	d22, err := darc.NewDarcFromProto(vs[0])
+	require.Nil(t, err)
+	require.False(t, d22.Equal(d2))
+	require.True(t, d22.Equal(s.darc))
+}
+
+func TestService_DarcEvolution(t *testing.T) {
+	s := newSer(t, 1, testInterval)
+	defer s.local.CloseAll()
+	defer closeQueues(s.local)
+
+	d2 := s.darc.Copy()
+	require.Nil(t, d2.EvolveFrom(s.darc))
+	resp := s.testDarcEvolution(t, *d2)
+
+	// parse the darc
+	require.True(t, resp.Proof.InclusionProof.Match())
+	_, vs, err := resp.Proof.KeyValue()
+	require.Nil(t, err)
+	d22, err := darc.NewDarcFromProto(vs[0])
+	require.Nil(t, err)
+	require.True(t, d22.Equal(d2))
+}
+
+func darcToCtx(t *testing.T, d2 darc.Darc, signer darc.Signer) ClientTransaction {
+	d2Buf, err := d2.ToProto()
+	require.Nil(t, err)
+	invoke := Invoke{
+		Command: "_evolve",
+		Args: []Argument{
+			Argument{
+				Name:  "darc",
+				Value: d2Buf,
+			},
+		},
+	}
+	instr := Instruction{
+		ObjectID: ObjectID{
+			DarcID:     d2.GetBaseID(),
+			InstanceID: ZeroNonce,
+		},
+		Nonce:  GenNonce(),
+		Index:  0,
+		Length: 1,
+		Invoke: &invoke,
+	}
+	require.Nil(t, instr.SignBy(signer))
+	ctx := ClientTransaction{
+		Instructions: []Instruction{instr},
+	}
+	return ctx
```

Comment:
Can we avoid using ctx to mean client transaction? ctx is well and truly a reserved word in my head for "context".